lib/views/pr-commit-view.js

```diff
@@ -37,14 +41,21 @@ export class PrCommitView extends React.Component {
     return moment(date).fromNow();
   }
 
+  openCommitDetailItem = () => {
+    return this.props.onBranch ? this.props.openCommit({sha: this.props.item.sha}) : null;
```

Comment:
We're gating this with `this.props.checkoutOp.why() === checkoutStates.CURRENT`, which is detected with:

https://github.com/atom/github/blob/0256c3dec21bc513857b61ce5e7f013b9425ced7/lib/controllers/issueish-detail-controller.js#L161-L180

Roughly, that reads "when the push upstream of the HEAD ref matches the head ref of the pull request."

The actual opening happens here:

https://github.com/atom/github/blob/0256c3dec21bc513857b61ce5e7f013b9425ced7/lib/controllers/issueish-detail-controller.js#L242-L246

Because the CommitDetailItem URI that's built includes the repository's workdir, that should ensure that the active repository is preserved in the new detail item, even if the Git tab's has changed. So you should be able to open commits in the same repository as the IssueishDetailController, not necessarily the same one as the Git tab.

I just verified that this works in dev mode 👍

Reply:
> So you should be able to open commits in the same repository as the IssueishDetailController, not necessarily the same one as the Git tab.

@smashwilson hmm doesn't the Git tab update based on the active pane item, and therefore the IssueishDetailController repository should match the Git tab repository?

And also, there's the added constraint of having to be on the currently checked out branch right? You can open commits in the same repository *if* the commit is on the checked out branch... at least that's the behavior that I'm observing.